upgrade_analysis/compare.py

```diff
@@ -128,6 +129,26 @@ def fieldprint(old, new, field, text, reprs):
             text += f" [{old['table']}]"
         if field == "relation":
             text += " [nothing to do]"
+        if field == "selection_keys":
+            old_selection_keys = old.get("selection_keys") or ""
```

Comment:
```suggestion
            old_selection_keys = old.get("selection_keys") or "[]"
```

This could avoid tests of type, dont you think ?

Reply:
but we want to do nothing in that case